src/functional/ffront/type_info.py

```diff
@@ -461,6 +461,22 @@ def return_type_scanop(
     )
 
 
+@return_type.register
+def return_type_scalarop(
+    callable_type: ct.ScalarOperatorType,
+    *,
+    with_args: list[ct.SymbolType],
+    with_kwargs: dict[str, ct.SymbolType],
+):
+    carry_dtype = callable_type.definition.returns
+    promoted_dims = promote_dims(
+        *(extract_dims(el) for arg in with_args for el in primitive_constituents(arg)),
+    )
+    return apply_to_primitive_constituents(
+        carry_dtype, lambda arg: ct.FieldType(dims=promoted_dims, dtype=cast(ct.ScalarType, arg))
+    )
+
```

Comment:
This is the same as `return_type_scanop`, do you maybe want to place this code in a function and avoid duplication?